kategory/src/main/kotlin/kategory/arrow/Function1.kt

```diff
@@ -6,20 +6,19 @@ fun <P, R> ((P) -> R).k(): Function1<P, R> =
         Function1(this)
 
 @Suppress("UNCHECKED_CAST")
-fun <R> Function1F<R>.ev(): FunctionInject<R> =
-        this as FunctionInject<R>
+fun <P, R> Function1F<R>.ev(): (P) -> R =
+        (this as Function1<P, R>).f
+
+fun <P, R> Function1F<R>.invoke(p: P): R =
+        this.ev<P, R>().invoke(p)
```

Comment:
Invoke to the rescue! It hides the intermediate ev() step from the callsites and provides inference for the compiler.